Dockerfile

```diff
@@ -0,0 +1,63 @@
+# syntax=docker/dockerfile:1.4
+
+ARG BASE_IMAGE="python:3.11-slim-bookworm"
+
+FROM ${BASE_IMAGE}
+
+ARG PYTORCH_INSTALL_ARGS=""
+ARG EXTRA_ARGS=""
+ARG USERNAME="comfyui"
+ARG USER_UID=1000
+ARG USER_GID=${USER_UID}
+
+RUN \
+	--mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+	--mount=target=/var/cache/apt,type=cache,sharing=locked \
+	set -eux; \
+		apt-get update; \
+		apt-get install -y --no-install-recommends \
+			git \
+			git-lfs \
+ 			rsync
+
+RUN set -eux; \
+	groupadd --gid ${USER_GID} ${USERNAME}; \
+	useradd --uid ${USER_UID} --gid ${USER_GID} -m ${USERNAME}
```

Comment:
Good point. Maybe I will remove the cache mounts for the sake of simplicity. The apt packages are not changing often anyway.

Reply:
As for SHELL and heredocs, I decided against using those because all docker-provided Dockerfiles seem to avoid them as well so far. I tried to mimic the best practices from the python Dockerfiles, IIRC.

Reply:
In another project (`uv`) where I'm contributing a `Dockerfile` the maintainers there were also not too fond of the cache mounts despite the benefits as it adds a bunch of noise/complexity that makes them uncomfortable to take on the burden of maintenance going forward. This is for rust where the cache mounts is a bit of a dance to do right 😓 

Regarding `SHELL`, I think it's mostly for similar reasons. That said, you'd probably want to unset those options once finalizing the image so it doesn't have an impact on runtime IIRC. Choosing to ditch it for the few `RUN` in the file where it'd be relevant is fine 👍 

HereDoc syntax is not that complicated `<<EOF` to mark the start, `EOF` to mark the end, multi-line content inbetween without having the noise of `&&` + `\` sounds like a win to me for readability/maintenance and I'm sure maintainers can understand that concept, barely a concern IMO, but I usually get pushback from it when contributing/improving `Dockerfile`s to projects 🤷‍♂️

Reply:
Sorry for the intrusion here.
Yeah, I've never seen mount apt cache and I've read quite a few Dockerfile till now.

Didn't know about HereDoc support. All in for it, for me it is also a win! https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/